qunetsim/components/host.py

```diff
@@ -924,6 +925,83 @@ def send_ghz(self, receiver_list, q_id=None, await_ack=False, no_ack=False, dist
             return q_id, ret
         return q_id
 
+    def send_w(self, receiver_list, q_id=None, await_ack=False, no_ack=False, distribute=False):
+        """
+        Share W state with all receiver ids in the list. W state is generated
+        locally.
+
+        Args:
+            receiver_list (list): A List of all Host IDs with which a W state
+                                  should be shared.
+            q_id (str): The ID of the W qubits
+            await_ack (bool): If the sender should await an ACK from all receivers
+            no_ack (bool): If this message should not use any ACK and sequencing.
+            distribute (bool): If the sender should keep part of the W state, or just
+                               distribute one
+        Returns:
+            (str, bool): Qubit ID of the shared W and ACK status
+        """
+
+        q_list: list[Qubit] = []
+        n = len(receiver_list) + (0 if distribute else 1)
+
+        def f_gate(i, j, k):
+            theta = math.acos(math.sqrt(1/(n-k+1)))
```

Comment:
We're using a single space between mathematical operations: eg `a+b` -> `a + b`

Reply:
ok, I'll update that